### PR TITLE
Added params.till to binance.fetchOHLCV

### PR DESCRIPTION
**Pull request by samgermain**
2022-06-15T08:11:42.620Z
Node.js: v14.17.0
CCXT v1.87.43

```
% binance fetchOHLCV ADA/USDT 1h 1654819200000 undefined '{"till": 1654905600000}'
binance.fetchOHLCV (ADA/USDT, 1h, 1654819200000, , [object Object])
2022-06-15T08:11:44.957Z iteration 0 passed in 672 ms

1654819200000 | 0.6323 | 0.6344 | 0.6129 | 0.6219 | 15095842.4
...
1654905600000 | 0.5744 | 0.5834 | 0.5725 | 0.5824 |    6446847
25 objects
```

```
% binanceusdm fetchOHLCV ADA/USDT 1h 1654819200000 undefined '{"till": 1654905600000}'
binanceusdm.fetchOHLCV (ADA/USDT, 1h, 1654819200000, , [object Object])
2022-06-15T08:11:55.510Z iteration 0 passed in 226 ms

1654819200000 |  0.632 | 0.6342 | 0.6114 | 0.6216 |  90372253
...
1654905600000 | 0.5738 |  0.583 | 0.5718 | 0.5819 |  35082125
25 objects
```

```
% binancecoinm fetchOHLCV ADA/USD 1h 1654819200000 undefined '{"till": 1654905600000}' 
binancecoinm.fetchOHLCV (ADA/USD, 1h, 1654819200000, , [object Object])
2022-06-15T08:12:15.643Z iteration 0 passed in 707 ms

1654819200000 | 0.63149 | 0.63356 | 0.61043 |   0.621 |  805764
...
1654905600000 | 0.57333 | 0.58233 | 0.57138 | 0.58139 |  242960
25 objects
```

```
% binanceusdm fetchOHLCV ADA/USDT 1h 1654819200000 undefined '{"till": 1654905600000, "price": "mark"}' | condense
binanceusdm.fetchOHLCV (ADA/USDT, 1h, 1654819200000, , [object Object])
2022-06-15T08:14:30.965Z iteration 0 passed in 243 ms

1654819200000 | 0.63201028 |     0.6341 | 0.61296974 |  0.6214461 | 0
...
1654905600000 | 0.57386208 |      0.583 | 0.57203525 | 0.58215502 | 0
25 objects
```

```
binancecoinm.fetchOHLCV (ADA/USD, 1h, 1654819200000)
2022-06-15T08:17:02.049Z iteration 0 passed in 727 ms
1654819200000 | 0.63149 | 0.63356 | 0.61043 |   0.621 |  805764
...
1655280000000 | 0.45569 |  0.4564 | 0.44886 | 0.45068 |  445645
129 objects
```

### CODE REVIEW

**Comment by kroitor** [comment]
@frosty00 in fact, `till` is a counterpart of `since` that is unified in some exchanges, since `endTime` is exchange specific and ccxt has not unified `startTime` for `endTime` to make any sense.

**Comment by frosty00** [comment]
why is it till and not untill

**Comment by samgermain** [comment]
> why is it till and not untill

I tried to set it to `until` initially, but it got changed to `till` and now `till` is in a bunch of places

**Comment by frosty00** [comment]
to me this is a till https://homestead-honey.com/wp-content/uploads/2014/05/Till2.jpg

**Comment by kroitor** [comment]
I think the `since` + `limit` design itself is not ideal, we can do better. I will propose a solution to this.

**Comment by frosty00** [comment]
imo until is the correct grammatical form

**Comment by frosty00** [comment]
in this case we should move everything into the params and make them the second arg

**Comment by samgermain** [comment]
> in this case we should move everything into the params and make them the second arg

what do you mean

**Comment by samgermain** [comment]
> > why is it till and not untill
> 
> Idk, for the same reason it's `since` and not `unsince` ? Just because it's shorter, i guess.

unsince isn't a word, till is slang for until I think

**Comment by kroitor** [comment]
@samgermain i agree, we should not mess with the signature of fetchOHLCV, but the goal is not so much about removing `till`, as it is about unifying the pagination, it can have different "allowed ranges" and can be id-based, date-based, or page/cursor based, so we will need some kind of pagination interface to cover all that in a uniform and convenient way. I will try to propose a better design for pagination as soon as I have a chance. In the meantime, we can rename `till` > `until`.

File: js/binance.js [hunk 1]
```diff
@@ -2316,7 +2319,9 @@ module.exports = class binance extends Exchange {
             // It didn't work before without the endTime
             // https://github.com/ccxt/ccxt/issues/8454
             //
-            if (market['inverse']) {
+            if (until !== undefined) {
```
Comment (frosty00) [inline js/binance.js:2322]: can we rewrite this so we only check once

**Pull request merged**